File: frontend/src/utils/playerName.js
// A deleted account's finished games belong to an anonymous user whose id starts with
// this prefix (zapzap-rust/src/domain/entities/user.rs, DELETED_USER_ID_PREFIX).
export const DELETED_USER_ID_PREFIX = 'deleted-';

export const DELETED_PLAYER_NAME = 'Joueur supprimé';

/** The name to show for a player of a finished game */
export const playerName = (userId, username) =>
  typeof userId === 'string' && userId.startsWith(DELETED_USER_ID_PREFIX)
    ? DELETED_PLAYER_NAME
    : username;

File: frontend-flutter/lib/utils/player_name.dart
import '../l10n/app_localizations.dart';

/// The id prefix of a deleted player. When a player deletes their account,
/// the backend moves their finished games to a stand-in account whose id
/// starts with it (`zapzap-rust/src/application/auth/delete_account.rs`), so
/// the other players' history keeps the game.
const deletedUserIdPrefix = 'deleted-';

/// Whether [userId] is such a stand-in.
bool isDeletedUser(String? userId) =>
    userId != null && userId.startsWith(deletedUserIdPrefix);

/// The name to show for a player of the history: « Joueur supprimé » /
/// "Deleted player" for a deleted account, else [username].
String playerName(AppLocalizations l10n, String? userId, String username) =>
    isDeletedUser(userId) ? l10n.deletedPlayer : username;
